docs/guides/define_agent.md

```diff
@@ -0,0 +1,192 @@
+              tendermint_com_url: ${TENDERMINT_COM_URL:str:http://localhost:8080}
+        ---
+        public_id: valory/abci:0.1.0
+        type: connection
+        config:
+          target_skill_id: valory/hello_world_abci:0.1.0
+          host: ${ABCI_HOST:str:localhost}
+          port: ${ABCI_PORT:int:26658}
+          use_tendermint: ${ABCI_USE_TENDERMINT:bool:false}
+        ---
+        public_id: valory/ledger:0.19.0
+        type: connection
+        config:
+          ledger_apis:
+            ethereum:
+              address: ${CONNECTION_LEDGER_CONFIG_LEDGER_APIS_ETHEREUM_ADDRESS:str:http://localhost:8545}
+              chain_id: ${CONNECTION_LEDGER_CONFIG_LEDGER_APIS_ETHEREUM_CHAIN_ID:int:31337}
+              poa_chain: ${CONNECTION_LEDGER_CONFIG_LEDGER_APIS_ETHEREUM_POA_CHAIN:bool:false}
+              default_gas_price_strategy: ${CONNECTION_LEDGER_CONFIG_LEDGER_APIS_ETHEREUM_DEFAULT_GAS_PRICE_STRATEGY:str:eip1559}
+        ---
+        public_id: valory/p2p_libp2p_client:0.1.0
+        type: connection
+        config:
+          nodes:
+          - uri: ${P2P_URI:str:acn.staging.autonolas.tech:9005}
+            public_key: ${P2P_PUBLIC_KEY:str:02d3a830c9d6ea1ae91936951430dee11f4662f33118b02190693be835359a9d77}
+          - uri: ${P2P_URI:str:acn.staging.autonolas.tech:9006}
+            public_key: ${P2P_PUBLIC_KEY:str:02e741c62d706e1dcf6986bf37fa74b98681bc32669623ac9ee6ff72488d4f59e8}
+        cert_requests:
+        - identifier: acn
+          ledger_id: ethereum
+          message_format: '{public_key}'
+          not_after: '2023-01-01'
+          not_before: '2022-01-01'
+          public_key: ${P2P_PUBLIC_KEY:str:02d3a830c9d6ea1ae91936951430dee11f4662f33118b02190693be835359a9d77}
+          save_path: .certs/acn_cosmos_9005.txt
+        - identifier: acn
+          ledger_id: ethereum
+          message_format: '{public_key}'
+          not_after: '2023-01-01'
+          not_before: '2022-01-01'
+          public_key: ${P2P_PUBLIC_KEY:str:02e741c62d706e1dcf6986bf37fa74b98681bc32669623ac9ee6ff72488d4f59e8}
+          save_path: .certs/acn_cosmos_9006.txt
+        is_abstract: true
+        ```
+
+3. **Create an entry for your agent in the local registry.** Add the corresponding entry to the local registry index file (`./packages/packages.json`). You must add the entry to the `dev` section, because it is a component being developed by you. You can use a placeholder for its hash value, as it will be corrected afterwards:
+
+    ```json
+    {
+      "dev": {
+        "agent/your_name/your_agent/0.1.0": "bafybei0000000000000000000000000000000000000000000000000000",
+        (...)
+      },
+      "third_party": {
+        (...)
+      }
+    }
+    ```
+
+    Update the package hashes. The command below will correct any hash mismatch in the `aea-config.yaml` file, as well as in the local registry index file (`./packages/packages.json`):
+
+    ```bash
+    autonomy packages lock
+    ```
```

Comment:
This was VERY confusing.

I got an error first time I tried. I tried to figure it out for a few minutes. Then deleted my workspace and started all over again.

Then, after following all the steps to the letter, I got the error:

Error: Found a package which is not listed in the `packages.json` with package id (agent, your_name/your_agent:0.1.0)

which I fixed by adding the corresponding entry to `packages.json`.

Then, I got the error 

Error: Found a package which is not listed in the `packages.json` with package id (skill, academy2/your_fsm_app:0.1.0)

which I fixed by adding the corresponding entry to `packages.json` and changing 'your_name' by the name I chose in the 'aea-config.yaml' I copied from the guide.

Reply:
I expected you to have these errors. Both of them come from the framework not  managing the local registry as expected (e.g., not adding the scaffolded FSM app to the packages). Hopefully this is resolved in 0.10.0.

Reply:
This is related to this issue: 
https://github.com/valory-xyz/open-autonomy/issues/1894

Also, will add pointers in the document to indicate the user to ensure the correct path.